client/src/pages/Budget.page.jsx

```diff
@@ -1,22 +1,160 @@
 import React from 'react';
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+import AddIcon from '@mui/icons-material/Add';
 
-import { ActionHeader, Card, Page } from 'ui';
-import { Grid } from '@mui/material';
+import {
+  ActionHeader,
+  Button,
+  Card,
+  ColorBox,
+  Error,
+  Loader,
+  LocalizedDate,
+  Money,
+  NoContent,
+  Page,
+  Table,
+} from 'ui';
+import { Box, Grid } from '@mui/material';
+import { BudgetService } from 'api';
 
 export const BudgetPage = () => {
+  const queryClient = useQueryClient();
+
+  const fetchBudget = () => {
+    return BudgetService.findAll();
+  };
+
+  const deleteBudget = (budgetsToRemove) => {
+    return BudgetService.remove({ ids: budgetsToRemove });
+  };
+
+  const { isLoading, data, isError, isFetching, isSuccess } = useQuery(
+    'budgetData',
+    fetchBudget,
```

Comment:
BudgerService może zostać użyty bezpośrednio w useQuery, nie jest koniecznie zamykanie go w funkcji. 

```suggestion
    BudgetService.findAll,
```